frontend/src/diagram/sprotty/convertDiagram.tsx

```diff
@@ -87,13 +89,19 @@ export const convertDiagram = (gqlDiagram: GQLDiagram, httpOrigin: string, readO
   diagram.targetObjectId = targetObjectId;
   diagram.features = createFeatureSet([hoverFeedbackFeature, viewportFeature]);
 
-  nodes.map((node) => convertNode(node, httpOrigin, readOnly, autoLayout)).map((node) => diagram.add(node));
+  nodes.map((node) => convertNode(diagram, node, httpOrigin, readOnly, autoLayout));
```

Comment:
because a child has to be added to a parent that is directly or indirectly under a graph root. So it up to the child to immediately add itself to its parent before adding its own children